### PR TITLE
Login Flow, APIs Authentication and Cookies Removal

### PR DESCRIPTION
**Pull request by irfanmurtaza-el**
- Login Page is Designed
- Home Page UI is Updated
- Document Page UI is Updated
- Removed Unnecessary Menu links and Icons
- Implement Auth Context
- Integrated AWS Congnito SDK on Frontend for Only Login
- Implemented Cognito Access Token Verification Process in APIs
- Delete Cookies Dependencies from Front-end and Back-end Side (POST Dpcument, GET Documents List, DELETE Document APIs are made secure)

### CODE REVIEW

File: context/credentials-context.tsx [hunk 1]
```diff
@@ -5,16 +5,16 @@ import cookies from "js-cookie"
...
-  pineconeEnvironment: process.env.NEXT_PUBLIC_PINECONE_ENVIRONMENT,
-  pineconeBucket: process.env.PINECONE_BUCKET_NAME,
-  pineconeIndex: process.env.NEXT_PUBLIC_PINECONE_INDEX_NAME,
-  pineconeApiKey: process.env.NEXT_PUBLIC_PINECONE_API_KEY,
-  supabaseUrl: process.env.NEXT_PUBLIC_SUPABASE_URL,
-  supabaseKey: process.env.NEXT_PUBLIC_SUPABASE_KEY,
-  supabaseBucket: process.env.NEXT_PUBLIC_SUPABASE_BUCKET,
-  supabaseDatabaseUrl: process.env.NEXT_PUBLIC_DATABASE_URL,
-  supabaseDirectUrl: process.env.NEXT_PUBLIC_DIRECT_URL,
+  openaiApiKey: "",
```
Comment (naqi) [inline context/credentials-context.tsx:8]: if these are empty where are they initialized?

Comment (naqi) [reply context/credentials-context.tsx:8]: Do we need to keep this variable?

File: jwks.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (naqi) [inline jwks.json:1]: what is this for? should this be in repo?

**Review by naqi — CHANGES_REQUESTED** [review]

File: jwks.json (hunk 2 above)
Comment (irfanmurtaza-el) [reply jwks.json:1]: This is public JSON Web Key to verify access token, Here is the Reference link : https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-verifying-a-jwt.html

File: context/credentials-context.tsx (hunk 1 above)
Comment (irfanmurtaza-el) [reply context/credentials-context.tsx]: Deleted Context for cookies

**Review by amiryamin450 — APPROVED** [review]

**Pull request merged**